### DISCUSSION TITLE
onError, onSuccess and onFinish

### DISCUSSION THREAD

**Discussion by albertcito** [urql-graphql/urql#1106] [Q&A]
I would like to do it. But I don't know if it's possible:

```javascript
const [result, reexecuteQuery] = useQuery({
    query: TodosQuery,
    onError: (error) => console.log(error),
    onSuccess: (response) => console.log(response),
    onFinish: (response) => console.log(response),
});
```

**Comment by kitten** [comment]
What’s your use-case here? In general, we’d probably recommend either switching to programmatic queries here or using effect hooks, but it depends on what you’re trying to write 😅

**Comment by albertcito** [comment]
I was using Apollo and when it's done sometimes I sent a success notification when it's saved. Or in some cases I send an error notification when it's fail. It's depends of the view.

![gif-urql](https://user-images.githubusercontent.com/1768483/97867952-d3661e80-1cdc-11eb-9f5c-1bd9a10d9bdd.gif)

**Reply by kitten** [reply]
Oh, that’s for a mutation 😅 typically, except if it’s optimistic, we’d recommend to use the promise on execution for that: https://formidable.com/open-source/urql/docs/basics/mutations/#using-the-mutation-result

**Reply by albertcito** [reply]
lol, It was very late for me to wrote query instead mutation xD Thank you a lot for your help :)
